pkgs/servers/sql/postgresql/default.nix

```diff
@@ -98,7 +98,7 @@ let
       ++ lib.optionals jitSupport [ "--with-llvm" ];
 
     patches = [
-      ./patches/disable-resolve_symlinks.patch
+      (if atLeast "16" then ./patches/disable-normalize_exec_path.patch else ./patches/disable-resolve_symlinks.patch)
```

Comment:
Note that this line is not moved/the order for patches isn't changed, to avoid spurious rebuilds on other versions. If it's less readable I'm fine with changing it, though.